proof-systems/src/darlin/pcd/simple_marlin.rs

```diff
@@ -23,6 +23,143 @@ use std::marker::PhantomData;
+    segment_size: usize,
+    max_domain_h_size: usize,
+    max_domain_k_size: usize,
+) -> bool
+{
+    let num_constraints = vk.index_info.num_constraints.next_power_of_two();
+    let num_variables = vk.index_info.num_variables.next_power_of_two();
+    let domain_h_size = std::cmp::max(num_constraints, num_variables);
+    let domain_k_size = vk.index_info.num_non_zero.next_power_of_two();
+    let num_segs = (domain_k_size as f64/segment_size.next_power_of_two() as f64).ceil() as usize;
+
+    domain_h_size <= max_domain_h_size.next_power_of_two() &&
+        domain_k_size <= max_domain_k_size.next_power_of_two() &&
+        vk.index_comms.iter().all(|comm| comm.comm.len() <= num_segs)
+}
+
+/// Check that proof is consistent with specified vk, min_segment_size, max_segment_size,
+/// max_domain_h_size and max_domain_k_size
+pub fn is_proof_consistent_with<G: AffineCurve, D: Digest>(
+    proof: &MarlinProof<G, D>,
+    vk: &MarlinVerifierKey<G::ScalarField, InnerProductArgPC<G, D>>,
+    min_segment_size: usize,
+    max_segment_size: usize,
+    max_domain_h_size: usize,
+    max_domain_k_size: usize,
+) -> bool
+{
+    let segment_size: usize = 1 << proof.pc_proof.proof.l_vec.len();
+    let zk_bound = if proof.pc_proof.proof.hiding_comm.is_some() { 1 } else { 0 };
+    let w_z_a_b_segs = ((max_domain_h_size + 2 * zk_bound) as f64/segment_size as f64).ceil() as usize;
+    let t_segs = ((max_domain_h_size as f64/segment_size as f64)).ceil() as usize;
+    let z_1_segs = ((max_domain_h_size + 3 * zk_bound) as f64/segment_size as f64).ceil() as usize;
+    let h_1_segs = ((2 * max_domain_h_size + 4 * zk_bound - 2) as f64/segment_size as f64).ceil() as usize;
+    let z_2_segs = (max_domain_k_size as f64/segment_size as f64).ceil() as usize;
+    let h_2_segs =  ((3 * max_domain_k_size - 3) as f64/segment_size as f64).ceil() as usize;
+
+    is_vk_consistent_with(vk, segment_size, max_domain_h_size, max_domain_k_size) &&
+        segment_size >= min_segment_size &&
+        segment_size <= max_segment_size &&
+        proof.commitments[0][0].comm.len() <= w_z_a_b_segs && // w poly
+        proof.commitments[0][1].comm.len() <= w_z_a_b_segs && // z_a poly
+        proof.commitments[0][2].comm.len() <= w_z_a_b_segs && // z_b poly
+        proof.commitments[1][0].comm.len() <= t_segs && // t poly
+        proof.commitments[1][1].comm.len() <= z_1_segs && // z_1 poly
+        proof.commitments[1][2].comm.len() <= h_1_segs && // h_1 poly
+        proof.commitments[2][0].comm.len() <= z_2_segs && // z_2 poly
+        proof.commitments[2][1].comm.len() <= h_2_segs // h_2 poly
+}
+
+/// Given segment_size, density, zk, proof_type, return the maximum value of |H| s.t. proof size will be <= max_proof_size
+/// and the corresponding proof size and vk size
+pub fn compute_max_domain_h_size(
+    segment_size: usize,
+    density: usize,
+    zk: bool,
+    max_proof_size: usize,
+    proof_type: &str
+) -> (usize, usize, usize)
```

Comment:
I'd still opt for a ``compute_max_constraints_and_variables()`` instead, as it is might be more informative in practice:
Given the density, a circuit might have more constraints than the one returned by the function. The reason for this is that the function applies the density to a power of two, which is only realistic if the circuit does in fact use all "space" given by a two-adic domain. In practice, the constraint system is padded, but this padding does not affect the non-zero entries hence the domain K needed.